src/v2/cookbook/unit-testing-vue-components.md

```diff
@@ -0,0 +1,234 @@
+
+## Details about the Value
+
+#### Why test?
+Component unit tests have lots of great points:
+- Provide documentation on how the component should behave
+- Save time over testing manually
+- Reduce bugs in new features
+- Improve design
+- Facilitate refactoring
+
+The benefits of automated testing are great and it what allows large teams of developers to maintain complex codebases. 
+
+#### Getting started
+
+[vue-test-utils](https://github.com/vuejs/vue-test-utils) is the official library for unit testing Vue components. The (vue-cli)[https://github.com/vuejs/vue-cli] webpack template comes with either Karma or Jest, both well supported test runners, and there are some (guides)[https://vue-test-utils.vuejs.org/en/guides/] in the `vue-test-utils` documentation.
+
+
+4. Discuss why this may be a compelling pattern. Links for reference are not required but encouraged.
+
+## Real-World Example
+
+Unit tests should be
+- Fast to run
+- Easy to understand
+- Only test a _single unit of work_
+
+Let's say we want to test this component. It shows a greeting, and asks for a username. If the username is less than seven letters, an error is displayed.
+
+```html
+<template>
+  <div>
+    <div class="message">
+      {{ message }}
+    </div>
+    Enter your username: <input v-model="usernane">
+    <div 
+      v-if="error"
+      class="error"
+    >
+      Please enter a username with at least seven letters.
+    </div>
+  </div>
+</template>
+
+<script>
+export default {
+  name: 'Foo',
+
+  data () {
+    return {
+      message: 'Hello World',
+      username: ''
+    }
+  },
+
+  computed: {
+    error () {
+      return this.username.length < 7
+    }
+  }
+}
+</script>
+```
+
+The things that we should test are:
+- is the `message` rendered?
+- if `error` is `true`, `<div class="error"`> should be present
+- if `error` is `false`, `<div class="error"`> should not be present
+
+And our first attempt at at test:
+
+```js
+import { shallow } from 'vue-test-utils'
+
+describe('Foo', () => {
+  it('renders a message and responds correctly to user input', () => {
+      const wrapper = shallow(Foo, {
+    data: {
+      message: 'Hello World',
+      username: ''
+    }
+  })
+
+  // see if the message renders
+  expect(wrapper.find('.message').text()).toEqual('Hello World')
+
+  // assert the error is rendered
+  expect(wrapper.find('.error').exists()).toBeTruthy()
+
+  // update the username and assert error is longer rendered
+  wrapper.setData({ username: 'Lachlan'  })
+  expect(wrapper.find('.error').exists()).toBeFalsey()
+  })
+})
+```
+
+There are some problems with the above:
+- a single test is making assertions about different things
+- difficult to tell the different states the component can be in, and what should be rendered
+
+The below example improves the test by:
+- only making one assertion per `it` block
+- having short, clear test descriptions
+- providing only the minimum data requires for the test
```

Comment:
Great, again, good job being clear and concise. The demo illustrates exactly this as well.